src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationParsingTests.cs

```diff
@@ -7458,16 +7437,183 @@ public void TestNullCheckedArgWithTrailingSpace()
             }
         }
 
-        [Fact(Skip = "PROTOTYPE(BangBang)")]
+        [Fact]
         public void TestNullCheckedArgWithTrailingNewLine()
         {
-            UsingStatement(@"void M(string name!=
-null) { }", options: TestOptions.RegularPreview, expectedErrors: new DiagnosticDescription[]
-            { 
-                // (1,19): error CS8712: Space required between '!' and '=' here.
-                    // void M(string name!=
-                Diagnostic(ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals, "!=").WithLocation(1, 19)
-            });
+            UsingStatement(@"void M(string name!!=null) { }", options: TestOptions.RegularPreview, expectedErrors: new DiagnosticDescription[0]);
+            N(SyntaxKind.LocalFunctionStatement);
+            {
+                N(SyntaxKind.PredefinedType);
+                {
+                    N(SyntaxKind.VoidKeyword);
+                }
+                N(SyntaxKind.IdentifierToken, "M");
+                N(SyntaxKind.ParameterList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.Parameter);
+                    {
+                        N(SyntaxKind.PredefinedType);
+                        {
+                            N(SyntaxKind.StringKeyword);
+                        }
+                        N(SyntaxKind.IdentifierToken, "name");
+                        N(SyntaxKind.ExclamationExclamationToken);
+                        N(SyntaxKind.EqualsValueClause);
+                        {
+                            N(SyntaxKind.EqualsToken);
+                            N(SyntaxKind.NullLiteralExpression);
+                            {
+                                N(SyntaxKind.NullKeyword);
+                            }
+                        }
+                    }
+                    N(SyntaxKind.CloseParenToken);
+                }
+                N(SyntaxKind.Block);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+        }
+
+        [Fact]
+        public void TestNullCheckedArgWithSpaceInbetween()
+        {
+            UsingStatement(@"void M(string name! !=null) { }", options: TestOptions.RegularPreview,
```

Comment:
Consider leaving a prototype comment to add the same range of single `!` tests you added on the lambda side.

Reply:
Same as above.